heart/heartbeat.py

```diff
@@ -471,11 +490,11 @@ def on_msg(self, ct, msg):
 				d["address"] = str(self.addr)
 				d["port"], d["players_current"], d["players_max"] = struct.unpack("<HHH", msg[:6])
 				msg = msg[6:]
-				d["name"] = stripnul(msg[:30])
+				d["name"] = msg[:30].strip("\x00")
```

Comment:
I totally missed this.
I think the confusion comes from the fact that `stripnul` doesn't simply strip null in this sense, as the name suggests, but instead returns everything before the first null.

	>>> stripnul("foo")
	'foo'
	>>> stripnul("foo\x00")
	'foo'
	>>> stripnul("foo\x00\x00")
	'foo'
	>>> stripnul("foo\x00bar")
	'foo'
	>>> stripnul("foo\x00bar\x00")
	'foo'